src/MagnumPlugins/BasisImporter/Test/BasisImporterTest.cpp

```diff
@@ -952,6 +1071,42 @@ void BasisImporterTest::rgbaUncompressedUastc() {
         (DebugTools::CompareImageToFile{_manager, 4.75f, 0.576f}));
 }
 
+void BasisImporterTest::rgbaUncompressedHdr() {
+    auto&& data = FileTypeData[testCaseInstanceId()];
+    setTestCaseDescription(data.name);
+
+    #if BASISD_LIB_VERSION < 150
+    CORRADE_SKIP("Current version of Basis doesn't support HDR.");
+    #endif
+
+    Containers::Pointer<AbstractImporter> importer = _manager.instantiate("BasisImporterRGBA16F");
+    /* See the comment above this variable for more details. The
+       rgbaf.{basis,ktx2} are both encoded with Y up in convert.sh but the KTX
+       files are missing the KTXorientation metadata, causing a warning. */
+    if(data.hasMissingOrientationMetadata)
+        importer->configuration().setValue("assumeYUp", true);
+    CORRADE_VERIFY(importer->openFile(Utility::Path::join(BASISIMPORTER_TEST_DIR,
+        "rgbaf"_s + data.extension)));
+    CORRADE_COMPARE(importer->image2DCount(), 1);
+
+    Containers::Optional<Trade::ImageData2D> image = importer->image2D(0);
+    CORRADE_VERIFY(image);
+    CORRADE_COMPARE(image->flags(), ImageFlags2D{});
+    CORRADE_VERIFY(!image->isCompressed());
+    CORRADE_COMPARE(image->format(), PixelFormat::RGBA16F);
+    CORRADE_COMPARE(image->size(), (Vector2i{63, 27}));
+
+    if(_manager.loadState("AnyImageImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("AnyImageImporter plugin not found, cannot test contents");
+    if(_manager.loadState("OpenExrImporter") == PluginManager::LoadState::NotFound)
+        CORRADE_SKIP("OpenExrImporter plugin not found, cannot test contents");
+
+    CORRADE_COMPARE_WITH(*image,
+        Utility::Path::join(BASISIMPORTER_TEST_DIR, "rgba-63x27.exr"),
+        /* There are moderately significant compression artifacts */
+        (DebugTools::CompareImageToFile{_manager, 0.535f, 0.12f}));
```

Comment:
Todo - this is incorrect, because it compares the source file with varying alpha to the transcoded file with alpha always set to 1. The error thresholds are *way* too high.

Reply:
You can slice just the RGB part from the file. Something like `image->pixels<Color4>().slice(&Color4::rgb)`  instead of `*image` should work, and then for the other side you can configure the importer to skip the alpha channel by setting `a=` in the config, which will not even import the alpha channel.

I think, at least.

Reply:
I tried the slice magic but it doesn't seem to work on a const view. Only with `mutablePixels()`. Is this an overload issue?

Reply:
Huh. What does it say?

Reply:
It says that it can't find a suitable overload. But I figured it out, `Color4<T>::rgb() const` (same for `Vector4`) return a value, not a reference. Will just do it manually then ☺️

Reply:
I actually ended up using `mutablePixels()` with a comment explaining why, hope that's no crime in the eyes of the const-ness committee

Reply:
> But I figured it out, `Color4<T>::rgb() const` (same for `Vector4`) return a value, not a reference.

Ahh, damn, that's because of the `constexpr` -- it seems that having the accessor constexpr was more valuable than having it sliceable. Sigh.

Yeah, `mutablePixels()` is fine. Or you could `arrayCast<const Color3h>()`, that'd achieve the same as the slice.